### DISCUSSION TITLE
Can I build an authentication package/app to share user data and auth across all apps?

### DISCUSSION THREAD

**Discussion by theotarr** [vercel/turborepo#825] [Help] [answered]
I am building a turborepo with four Next.js apps that all require user login and authentication. I am using Prisma with Postgres and Next Auth for authentication and I would like to build one auth app that handles all authentication. Basically, a custom wrapper for Next Auth that takes in a redirect callback URL and redirects the user to that URL after being authenticated. Are there any examples or resources to do this? I this wrapper could and should probably be a custom express server that takes these params and creates the necessary Next Auth redirect.

**Comment by graup** [comment]
I don't understand what this has to do with Turborepo. Yes, you can build any package you want and have Turborepo build it.

**Reply by isaaccustodio** [reply]
How can you understand something you don't even want to?

**Comment by isaaccustodio** [comment, marked as answer]
> I am building a turborepo with four Next.js apps that all require user login and authentication. I am using Prisma with Postgres and Next Auth for authentication and I would like to build one auth app that handles all authentication. Basically, a custom wrapper for Next Auth that takes in a redirect callback URL and redirects the user to that URL after being authenticated. Are there any examples or resources to do this? I this wrapper could and should probably be a custom express server that takes these params and creates the necessary Next Auth redirect.

Yes you can build an authentication service package, and share data with any package you want, exporting it from your new auth workstation, and declaring in your app package with --> "<your-workstation-name>": "*"   ☺️

**Reply by Mac-DevOSYandex** [reply]
That's a great answer! How exactly!?! 
That's exactly what I am doing... 
I have a db/ under packages/ with Prisma ORM
Should I create a core/ and put all the Authjs@5 in there and connect it the prisma in the db/

There should be an example...

**Reply by pantharshit007** [reply]
did you succeed?
